src/SDK/Trace/Behavior/SpanExporterTrait.php

```diff
@@ -45,7 +56,55 @@ public function export(iterable $spans): int
             return SpanExporterInterface::STATUS_SUCCESS;
         }
 
-        return $this->doExport($spans); /** @phpstan-ignore-line */
+        // If retryPolicy is not set then just doExport once and return the status
+        if ($this->retryPolicy === null) {
+            self::logDebug('retry polict is null. Exporting the spans without retry');
+
+            return $this->doExport($spans); /** @phpstan-ignore-line */
+        }
+
+        while ($shouldRetry) {
+            // find the delay time before retry (it is 0 for the first attempt)
+            $delay = $this->retryPolicy->getDelay($attempt);
+            if ($attempt > 0) {
+                self::logDebug('Waiting for ' . $delay . ' mili seconds before retrying export');
+                $this->retryPolicy->delay($delay);
+                self::logDebug('Retrying span export for ' . $attempt . ' time');
+            }
+            $attempt++;
+
+            try {
+                $status = $this->doExport($spans);
+            } catch (Throwable $e) {
+                if ($e instanceof \Error) {
```

Comment:
Retrying on `Error` seems incorrect; can we remove the exception handling completely?
This does not set `$status` and will provide an incorrect value (the previous `$status`) to `RetryPolicy::shouldRetry()`.

Reply:
Done